### DISCUSSION TITLE
Mergind and shuffling two datasets

### DISCUSSION THREAD

**Discussion by a-ism256** [pyg-team/pytorch_geometric#3149] [Q&A] [answered]
Hello,

I'm wondering if there's a way to train on two datasets but not sequentially.

So let's say I have Dataset1 containing graphs A, B and C and Dataset2 containing graphs D and E. I would like to be able to generate a random sequence like B, D, A, C, E to train my model on.

Is there a way to currently do this without writing a third Dataset class that processes both sets of graphs?

Thanks.

**Comment by rusty1s** [comment, marked as answer]
You can use `torch.utils.data.ConcatDataset` for that, see [here](https://pytorch.org/docs/stable/data.html).

**Reply by a-ism256** [reply]
Alright I'll try it thanks! That was a really quick reply :smile:.
